cfme/cloud/flavor.py

```diff
@@ -107,12 +144,62 @@ class Flavor(BaseEntity, Taggable):
 
     name = attr.ib()
     provider = attr.ib()
+    ram = attr.ib()
```

Comment:
These are now required attributes, and the test is hitting a TypeError because its instantiating with only name/provider.

These attributes should be optional so we can instantiate with just name and provider.

```
provider = OpenStackProvider(endpoints={'default': <cfme.infrastructure.provider.openstac...ne, keystone_v3_domain_id='default', tenant_mapping=False, infra_provider=None)

    @pytest.fixture(scope='function')
    def zero_disk_flavor(provider):
        api_flv = provider.mgmt.api.flavors.create(fauxfactory.gen_alpha(), RAM, VCPUS, ZERO_DISK_SIZE)
        wait_for(provider.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600)
        zero_disk_flavor = provider.appliance.collections.cloud_flavors.instantiate(
>           api_flv.name, provider)

cfme/tests/openstack/cloud/test_flavors.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cfme/modeling/base.py:119: in instantiate
    return self.ENTITY.from_collection(self, *args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'cfme.cloud.flavor.Flavor'>
collection = FlavorCollection(filters={})
k = ('KdDEpdUHXb', OpenStackProvider(endpoints={'default': <cfme.infrastructure.provider.openstac...ne, keystone_v3_domain_id='default', tenant_mapping=False, infra_provider=None))
kw = {}

    @classmethod
    def from_collection(cls, collection, *k, **kw):
>       return cls(collection, *k, **kw)
E       TypeError: __init__() takes at least 9 arguments (4 given)

cfme/modeling/base.py:152: TypeError
```

Reply:
How could they be made optional ?
might be with default=None

Reply:
Just like this ```attr.ib(default=None)```